java/core/src/main/java/sleeper/core/util/ByteArray.java

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright 2022-2025 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.core.util;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+/**
+ * Class to provide wrapper object for primitive byte array into an object.
+ * Require as hashCode and equals requied for usage of type within a HashSet.
+ *
+ */
+public abstract class ByteArray implements Comparable<ByteArray> {
+
+    public static final ByteArrayComparator BYTE_ARRAY_COMPARATOR = new ByteArrayComparator();
+
+    /**
+     * Returns contents of object as primitive type.
+     *
+     * @return array details in primitive type
```

Comment:
I'm not sure what "details" or "in primitive type" means here. I'd just put "the byte array".